File: app/src/main/java/hibernate/v2/testyourandroid/model/GridItem.java
package hibernate.v2.testyourandroid.model;

/**
 * Created by himphen on 24/5/16.
 */

public class GridItem {
	private String mainText;
	private int mainImageId;
	private Class intentClass;
	private String actionType;

	public String getActionType() {
		return actionType;
	}

	public void setActionType(String actionType) {
		this.actionType = actionType;
	}

	public String getMainText() {
		return mainText;
	}

	public int getMainImageId() {
		return mainImageId;
	}

	public Class getIntentClass() {
		return intentClass;
	}

	public GridItem(String mainText, int mainImageId, Class intentClass) {
		this.mainText = mainText;
		this.mainImageId = mainImageId;
		this.intentClass = intentClass;
	}

	public GridItem(String mainText, int mainImageId, String actionType) {
		this.mainText = mainText;
		this.mainImageId = mainImageId;
		this.actionType = actionType;
	}
}

File: app/src/main/java/hibernate/v2/testyourandroid/model/GridItem.kt
package hibernate.v2.testyourandroid.model

/**
 * Created by himphen on 24/5/16.
 */
class GridItem(
    var text: String,
    var image: Int,
    var badge: Badge = Badge.NONE,
    var intentClass: Class<*>? = null,
    var action: Action? = null
) {
    enum class Badge {
        NONE,
        NEW,
        BETA,
    }

    enum class Action {
        HOME_RATE,
        HOME_LANGUAGE,
        HOME_DONATE,
        HOME_APP_BRAIN,
        APP_INFO_OPEN,
        APP_INFO_UNINSTALL,
        APP_INFO_SETTINGS,
        APP_INFO_PLAY_STORE,
    }
}